File: carbonaudit/src/main/java/com/carbonaudit/carbonaudit/service/AuditService.java
package com.carbonaudit.carbonaudit.service;

import com.carbonaudit.carbonaudit.dto.AuditDTO;

public interface AuditService {

    Object getPendingRecords();
    Object getAllAudits();

    Object approveRecord(
            Long auditId,
            AuditDTO auditDTO);

    Object rejectRecord(
            Long auditId,
            AuditDTO auditDTO);
}

File: carbon-audit-frontend/src/services/auditService.js
import api from "../utils/axiosConfig";

export const getPendingRecords = async () => {

  const response =
    await api.get(
      "/audit/pending"
    );

  return response.data;
};

export const getAllAudits = async () => {

  const response =
    await api.get(
      "/audit/all"
    );

  return response.data;
};

export const approveRecord =
  async (
    auditId,
    auditData
  ) => {

    const response =
      await api.put(
        `/audit/approve/${auditId}`,
        auditData
      );

    return response.data;
  };

export const rejectRecord =
  async (
    auditId,
    auditData
  ) => {

    const response =
      await api.put(
        `/audit/reject/${auditId}`,
        auditData
      );

    return response.data;
  };

const auditService = {

  getPendingRecords,

  getAllAudits,

  approveRecord,

  rejectRecord

};

export default auditService;